src/account/tests.py

```diff
@@ -1,3 +1,320 @@
+
+        self.client.logout()
+
+    def test_get_single_random_mod(self):
+        self.client.force_login(User.objects.get(username='mod'))
+        get_idx = Profile.objects.order_by('?').first().pk
+        url = ''.join([reverse('profile-list'), str(get_idx), '/'])
+        response = self.client.get(url)
+
+        profile = Profile.objects.get(pk=get_idx)
+        serializer = ProfileSerializer(profile)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data, serializer.data)
+
+        self.client.logout()
+
+    def test_get_single_own_user(self):
+        self.client.force_login(User.objects.get(username='user'))
+        get_idx = User.objects.get(username='user').pk
+        url = ''.join([reverse('profile-list'), str(get_idx), '/'])
+        response = self.client.get(url)
+
+        profile = Profile.objects.get(pk=get_idx)
+        serializer = ProfileSerializer(profile)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data, serializer.data)
+
+        self.client.logout()
+
+    def test_get_single_not_own_user(self):
+        self.client.force_login(User.objects.get(username='user'))
+        get_idx = User.objects.get(username='other_user').pk
+        url = ''.join([reverse('profile-list'), str(get_idx), '/'])
+        response = self.client.get(url)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        self.client.logout()
+
+    def test_get_single_non_exist_admin(self):
+        self.client.force_login(User.objects.get(username='admin'))
+        get_idx = Profile.objects.all().count() + 1
+        url = ''.join([reverse('profile-list'), str(get_idx), '/'])
+        response = self.client.get(url)
+
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+        self.client.logout()
+
+    # -------------------------------------------------------------------------
+    # PUT
+    # -------------------------------------------------------------------------
+
+    def test_put_single_admin(self):
+        self.client.force_login(User.objects.get(username='admin'))
+        put_idx = User.objects.all()[random.randint(0, User.objects.all().count() - 1)].pk
+        url = ''.join([reverse('profile-list'), str(put_idx), '/'])
+
+        data = {
+            "id": put_idx,
+            "user": {
+                "username": User.objects.get(pk=put_idx).username,
+                "email": fake_email,
+                "first_name": fake.text(max_nb_chars=20),
+                "last_name": fake.text(max_nb_chars=20),
+                "is_staff": False,
+                "is_active": True
+            },
+            "about_me": fake.text(max_nb_chars=50),
+            "is_score_visible": True,
+            "ranked": True
+        }
+
+        response = self.client.put(url, data, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
```

Comment:
Need to test if the changes took place in the database, just to be sure (with GET or field-by-field).
Field-by-field method:
```
self.assertEqual(User.objects.get(pk=put_idx).first_name, data['user']['first_name'])
...
```